File: leetcode/0451-sort-characters-by-frequency/solution.cpp
class Solution {
public:
    string frequencySort(string s) {
        map<int,vector<char>,greater<int>> count;
        sort(s.begin(),s.end());
        for (int i=0;i<s.size();i++){
            int cnt=1;
            while(i+1<s.size() && s[i]==s[i+1]){
                cnt++;
                i++;
            }
            count[cnt].push_back(s[i]);
        }
        string result="";
        for (const auto &pair : count) {
            int frequency = pair.first;
            const vector<char>& characters = pair.second;

        for (char c : characters) { // New loop: Go through each char with this frequency
            for (int i = 0; i < frequency; i++) {
                result += c;
                }
            }
        }
        return result;


    }
};

File: leetcode/0451-sort-characters-by-frequency/solution.py
class Solution(object):
    def frequencySort(self, s):
        count={}
        for i in s:
            if (i not in count.keys()):
                count[i]=1
            else:
                count[i]+=1
        sort_key=sorted(count.items(),key=lambda item:item[1],reverse=True)
        string=[]
        for i in sort_key:
            string.append(i[0]*i[1])
        return "".join(string)
        """
        :type s: str
        :rtype: str
        """
        
